### ISSUE TITLE
Mesh computation doesn't respect active mapping

### ISSUE THREAD

**Issue by philippotto** [scalableminds/webknossos#5560]
When enabling a mapping (either via json or agglomerate file), computing/loading a mesh doesn't respect that mapping.

**Comment by fm3** [comment]
Is this about the on-the-fly mapping computation or about loading a precomputed mesh?

**Comment by philippotto** [comment]
Good point. I think, this affects the following:

- on-the-fly meshes if the current mapping is a dynamic one (usually due to merger mode)
- precomputed meshes (maybe my assumption is wrong, but I don't think this is handled, right now?)

For on-the-fly meshes with persisted mappings, the feature should work (at least, the code suggests so).

**Comment by fm3** [comment]
Sounds about right, thanks! I think for precomputed meshes this could be solved by selecting the right meshfile file name to match the agglomerate file.

I don’t know enough about dynamic mappings to say how we can pass them to the back-end for this

**Comment by philippotto** [comment]
> Sounds about right, thanks! I think for precomputed meshes this could be solved by selecting the right meshfile file name to match the agglomerate file.

Yes, we probably need to know/persist that relation somehow. I also created https://github.com/scalableminds/webknossos/issues/5563 for this.

> I don’t know enough about dynamic mappings to say how we can pass them to the back-end for this

The dynamic mappings are usually rather small and could be sent to the back-end as JSON. The merger mode would be a special case where the front-end converts the skeletons to a dictionary. Since the back-end also has the skeletons we wouldn't need to send anything additionally; however, for the sake of simplicity, we could still send the dictionary to the back-end.

The problem with the dynamic mappings is rather that they are ... well, dynamic. If they change, the mesh should change, too, ideally. We could apply the mapping in the front-end (i.e., load the mesh of the unmapped ids and then map them afterwards). However, there are quite a bit of details which would need to figure out probably. I think, we should have a meeting about this at some point :)

**Comment by fm3** [comment]
What’s the state here now? I assume the code changed quite a bit since the last discussion? Can we narrow down the scope of what needs too be done here?

**Comment by philippotto** [comment]
I think, we have the following factors:
- hdf5 mapping, json mapping (persisted on disk), local mapping (e.g., merger mode)
- adhoc and precomputed meshing

I assume that most of these combinations work by now, but local/dynamic mappings are not accounted for. I updated the issue title/description accordingly. I think, the proper solution would be to handle the mapping locally in the front-end, but I question the priority, because we want to ditch the merger mode in the long run in favor of the proofreading tool.

**Comment by fm3** [comment]
Another error case we saw: in case the segment index is used (by the frontend) to determine which mesh chunks to load, an active (static) json mapping also appears to not be respected.

Compare https://github.com/scalableminds/webknossos/pull/9778#issuecomment-4934027866
